src/core/execute.c

```diff
@@ -3610,8 +3610,7 @@ void exec_context_done(ExecContext *c) {
 
         bind_mount_free_many(c->bind_mounts, c->n_bind_mounts);
 
-        if (c->cpuset)
-                CPU_FREE(c->cpuset);
+        c->cpuset = mfree(c->cpuset);
```

Comment:
hmm, so we should probably either use only libc's official allocation/free calls for CPU sets (i.e. CPU_FREE, CPU_ALLOC and friends), or only plan memory allocations (i.e. free/mfree, new/malloc/memdup/…), but not mix them, i.e. not CPU_FREE something that was allocated through memdup(), and not free() something that was allocated with CPU_ALLOC. Most likely it ultimately would just work, but it's still ugly since these are two independent sets of calls